### PR TITLE
feat: Allow multiple launches from single IBM instance

### PR DESCRIPTION
**Pull request by TheRealFalcon**
The default naming convention for new launches from the IBM cloud class used the same name for every launch from that instance. IBM cloud doesn't allow this, so ensure names are unique.

### CODE REVIEW

File: pycloudlib/ibm/cloud.py [hunk 1]
```diff
@@ -72,6 +73,7 @@ def __init__(
 
         api_key = api_key or self.config.get("api_key")
         authenticator = IAMAuthenticator(api_key)
+        self.instance_counter = itertools.count()
```
Comment (blackboxsw) [inline pycloudlib/ibm/cloud.py:76]: 1 for fun
```suggestion
        self.instance_counter = itertools.count(1)
```

File: pycloudlib/ibm/cloud.py [hunk 2]
```diff
@@ -350,7 +352,7 @@ def launch(
 
         vpc = vpc or self.vpc
         floating_ip_name = f"{name}-fi" if name else None
-        name = name or f"{self.tag}-vm"
+        name = name or f"{self.tag}-{next(self.instance_counter)}vm"
```
Comment (blackboxsw) [inline pycloudlib/ibm/cloud.py:355]:
```suggestion
        name = name or f"{self.tag}-vm{next(self.instance_counter)}"
```

**Review by blackboxsw — APPROVED** [review]
just bikesheds, LGTM either way. I know we are pressed for instance name length on some platforms, so avoiding excess hyphens may come into play at some point if we have datetimestamps in our tags. But looks good

**Review by a-dubs — APPROVED** [review]
Changes look good. So this is for the use case where you keep the same cloud object and try to launch multiple instances from it?

**Comment by TheRealFalcon** [comment]
> Changes look good. So this is for the use case where you keep the same cloud object and try to launch multiple instances from it?

Yep!

**Pull request merged**